### DISCUSSION TITLE
webp image format

### DISCUSSION THREAD

**Discussion by PippoApps** [airsdk/Adobe-Runtime-Support#2251] [Ideas]
Hello there,
would be great to have support for webp image format in AIR apps.
It allows to save a lot of space, especially when transparency is required, forcing the use of png.
Cheers

**Comment by teotigraphix** [comment]
https://developers.google.com/speed/webp/docs/api

`libwebapi` C library.

**Reply by PippoApps** [reply]
Cheers.
I would be much happier to WEBP.getBitmapData(ba:Bytearray) 😄

**Reply by marchbold** [reply]
We do have the WebP ANE available which can do a lot of this already: https://docs.airnativeextensions.com/docs/webp/parse-webp-data

**Reply by PippoApps** [reply]
Thank you.
I see there are many new useful ANEs I didn't know about.
Wonderful

**Reply by marchbold** [reply]
Glad they are useful! Docs need a bit of an update for that one, it's been around for a while, let me know if anything isn't clear.

**Reply by teotigraphix** [reply]
> Cheers. I would be much happier to WEBP.getBitmapData(ba:Bytearray) 😄

Of course! But I didn't know @marchbold already had that ANE, I was just implying, the lib is open for ANE implementation.

**Reply by PippoApps** [reply]
ahahah true :)
I do need a cross-platform solution for webp that works on devices and desktops... I guess I will stick to PNGs and use lower size/quality on Android...

**Reply by teotigraphix** [reply]
Wish I had the iOS experience with ANEs I could whip this up. I use my own anes on Windows and Android, so kind of type cast, no pun here.. :)

**Reply by PippoApps** [reply]
ahahah true... I build artworks that use a lot of audio-visual assets, on all major platforms it is not an issue, but on Android the store limitation is 150Mb - nothing if you use PNGs and hours of high quality sounds - and I don't want to rely on external bundles not to break the full cross-platform paradigm... not worth anyway...

**Reply by PippoApps** [reply]
thank you very much for your offer to help :)

**Reply by teotigraphix** [reply]
> external bundles not to break the full cross-platform paradigm... not worth anyway...

I have this constraint in my apps as well, it's a pain when you are trying to make it all jive and can't just use the OS framework stuff easy, been there many times..

**Reply by marchbold** [reply]
Wouldn't take much for me to port it to macOS, I can look into the windows implementation in a month or so but have a lot of work on right now.
